File: Tema1/ErrorRedondeo.py
class ErrorRedondeo:
    @staticmethod
    def calcular(valor_original, cifras_decimales):
        valor_redondeado = round(valor_original, cifras_decimales)
        return abs(valor_original - valor_redondeado)

    @staticmethod
    def calcular_porcentaje(valor_original, cifras_decimales):
        valor_redondeado = round(valor_original, cifras_decimales)
        return (abs(valor_original - valor_redondeado) / abs(valor_original)) * 100


if __name__ == "__main__":
    er = ErrorRedondeo.calcular(3.14159, 3)
    er_pct = ErrorRedondeo.calcular_porcentaje(3.14159, 3)
    print(f"Error de redondeo: {er}")
    print(f"Error de redondeo (%): {er_pct:.6f} %")

File: tema1/ErrorRedondeo.java
/**
 * Calcula el error absoluto y porcentual por redondeo a cierto número de decimales.
 */
public class ErrorRedondeo {
    /**
     * Calcula el error absoluto al redondear.
     * @param valorOriginal Valor real.
     * @param cifrasDecimales Cantidad de decimales a redondear.
     * @return Error absoluto.
     */
    public static double calcular(double valorOriginal, int cifrasDecimales) {
        double factor = Math.pow(10, cifrasDecimales); // escala para redondeo
        double valorRedondeado = Math.round(valorOriginal * factor) / factor;
        return Math.abs(valorOriginal - valorRedondeado);
    }

    /**
     * Calcula el error porcentual al redondear.
     * @param valorOriginal Valor real.
     * @param cifrasDecimales Decimales a redondear.
     * @return Porcentaje de error.
     */
    public static double calcularPorcentaje(double valorOriginal, int cifrasDecimales) {
        double factor = Math.pow(10, cifrasDecimales);
        double valorRedondeado = Math.round(valorOriginal * factor) / factor;
        return (Math.abs(valorOriginal - valorRedondeado) / Math.abs(valorOriginal)) * 100;
    }

    public static void main(String[] args) {
        double valor = 3.14159;
        double error = calcular(valor, 3); // redondear a 3 cifras decimales
        double errorPct = calcularPorcentaje(valor, 3);
        System.out.println("Error de redondeo: " + error);
        System.out.printf("Error de redondeo (%%): %.6f %%\n", errorPct);
    }
}
